client/dashboard/sites/settings-site-visibility/privacy-form.tsx

```diff
@@ -97,6 +118,12 @@ export function PrivacyForm( {
 	settings: SiteSettings;
 	mutation: UseMutationResult< Partial< SiteSettings >, Error, Partial< SiteSettings >, unknown >;
 } ) {
+	const { data: domains = [] } = useQuery( siteDomainsQuery( site.ID ) );
```

Comment:
Nit: since the query is ensured in the loader, we can use a more type-safe way

```suggestion
	const { data: domains } = useSuspenseQuery( siteDomainsQuery( site.ID ) );
```

Reply:
😮 TIL about `useSuspenseQuery()`.

It doesn't have a lot of documentation, but it looks very fancy. I am interested in making sure everything still "works" even if we deleted all the loaders. I feel the loaders should just be optimisations/UI improvements.

I tested to see how the loading behaves differently if the loader is missing. The loading is slightly different. With `useSuspenseQuery()` the secondary menu appears, disappears, and reappears while loading. I'm not a React suspense expert, but maybe it's because when suspense pauses the render, we haven't set it up properly to be able to pause rendering while keeping the secondary menu present.

The page does still load correctly though when the loader is missing. `useSuspenceQuery()` is something I'm interested in investigating in future PRs.